examples/terraform/gce/main.tf

```diff
@@ -132,7 +132,7 @@ resource "google_compute_forwarding_rule" "control_plane" {
 resource "google_compute_instance" "control_plane" {
   count = 3
 
-  name         = "${var.cluster_name}-control-plane-${count.index + 1}"
+  name         = "${var.cluster_name}-master-${count.index + 1}"
```

Comment:
When we have such behavior, i.e. the FQDN is way too long, we usually override the hostname. You can see #607 as an example, which should fix the issue you mention for GCP, but it was never merged because I never got E2E tests working (which was the purpose of the PR).

If you want to go with that approach, you can update your PR to set the hostname and I'll rebase later.

Reply:
Ok.. Which approach do you prefer? Any of the two workarounds is okay for me.

Reply:
I would go with the setting the hostname approach as we use it for some other providers too.

Reply:
Ok.. will wait for your change to be merged then. I also support the idea that the behavior should be consistent across the providers.

Reply:
#607 has been merged to fix this.